File: src/NominaPro.Domain/Entities/Usuario.cs
using NominaPro.Domain.Common;

namespace NominaPro.Domain.Entities;

public class Usuario : BaseEntity
{
    public string Nombre { get; set; } = string.Empty;

    public string Email { get; set; } = string.Empty;

    public string PasswordHash { get; set; } = string.Empty;

    public string Rol { get; set; } = "Consulta";
    public Guid EmpresaId { get; set; }

    public Empresa Empresa { get; set; } = null!;
}

File: frontend/src/types/usuario.ts
export type Usuario = {
  id: string;
  nombre: string;
  email: string;
  rol: string;
  empresaId: string;
  activo: boolean;
};

export type CreateUsuarioRequest = {
  nombre: string;
  email: string;
  password: string;
  rol: string;
  empresaId: string;
};

export type UpdateUsuarioRequest = {
  nombre: string;
  email: string;
  rol: string;
  activo: boolean;
};